web/src/hashchange.js

```diff
@@ -198,7 +213,7 @@ function do_hashchange_normal(from_reload) {
                 return false;
             }
             const narrow_opts = {
-                change_hash: false, // already set
+                change_hash: require_hash_change(hash),
```

Comment:
This would be important when channel links in left sidebar go to first topic. In that case, rather than calling `message_view.show`, `browser_history.go_to_location` is used, which would require you to update hash to attach `with` operator.

Reply:
Hmm, I wonder if we want to instead use a different type of parameter/check? 

Like, maybe `change_hash` should be rebranded to `add_history_entry`; I think what we probably want is the URL change to include `/with/` in the URL would be not add an extra history visible to the `back` button, and just be a change in the URL, whereas I think most other callers of this function with `change_hash: true` want a new history entry. 

If that fits your understanding, maybe a good next step on this detail would be a prep PR to rename `change_hash` to `add_history_entry` and maybe change a few other names in that code path to align with it.